tests/acceptance/helpers/ldapHelper.js

```diff
@@ -0,0 +1,216 @@
+const { client } = require('nightwatch-api')
+const ldap = require('ldap')
+const { join } = require('./path')
+const fs = require('fs-extra')
+
+const userHelper = require('./userSettings')
+
+exports.createClient = function (url) {
+  return new Promise((resolve, reject) => {
+    const ldapClient = ldap.createClient({
+      url: url || client.globals.ldap_url
+    })
+
+    ldapClient.bind(client.globals.ldap_base_dn, 'admin', err => {
+      if (err) {
+        reject(err)
+      }
+    })
+
+    ldapClient.add('ou=TestUsers,dc=owncloud,dc=com', {
+      objectclass: ['top', 'organizationalUnit'],
+      ou: 'TestUsers'
+    }, (err) => {
+      if (err) {
+        console.log('Users OU already exists.')
+      }
+      ldapClient.add('ou=TestGroups,dc=owncloud,dc=com', {
+        objectclass: ['top', 'organizationalUnit'],
+        ou: 'TestGroups'
+      }, (err) => {
+        if (err) {
+          console.log('Groups OU already exists.')
+        }
+        resolve(ldapClient)
+      })
+    })
+  })
+}
+
+exports.createEntry = function (client, dn, entry) {
+  return new Promise((resolve, reject) => {
+    client.add(dn, entry, function (err) {
+      if (err) {
+        reject(err)
+      }
+      console.log('Created LDAP entry')
+      resolve()
+    })
+  })
+}
+
+exports.cleanup = function (client) {
+  return new Promise((resolve, reject) => {
+    client.del('ou=TestGroups,dc=owncloud,dc=com', function (err) {
+      if (err) {
+        reject(err)
+      }
+      resolve()
+    })
+  }).then(() => {
+    client.del('ou=TestUsers,dc=owncloud,dc=com', function (err) {
+      if (err) {
+        Promise.reject(err)
+      }
+      return Promise.resolve()
+    })
+  })
+}
+
+exports.getNewUID = function (ldapClient) {
+  const userCn = 'ou=TestUsers,dc=owncloud,dc=com'
+  let uid = 1
+  return new Promise((resolve, reject) => {
+    ldapClient.search(
+      userCn,
+      { scope: 'sub', attributes: ['uidNumber'] },
```

Comment:
does this get always the last (highest) id on the server?

Reply:
Yes, It checks all the users and gets the new id from them (The new id will be max id + 1 )